evalml/automl/data_splitters/training_validation_split.py

```diff
@@ -41,5 +42,5 @@ def split(self, X, y=None):
             Returns:
                 list: indices to split data into training and test set
         """
-        train, test = train_test_split(X.index, test_size=self.test_size, train_size=self.train_size, shuffle=self.shuffle, stratify=self.stratify, random_state=self.random_state)
+        train, test = train_test_split(np.arange(X.shape[0]), test_size=self.test_size, train_size=self.train_size, shuffle=self.shuffle, stratify=self.stratify, random_state=self.random_state)
```

Comment:
Ok, yep, I think this makes sense and aligns with sklearn's behavior.

The following
```python
from sklearn.model_selection import KFold
N = 11000
X = pd.DataFrame({'col1': np.arange(0, N)}, index=np.arange(20000, 20000 + N))
sk_splits = list(KFold(2, random_state=0).split(X))
sk_splits[0]
```
outputs splits generated in terms of the row number, ignoring the custom index:
```python
(array([ 5500,  5501,  5502, ..., 10997, 10998, 10999]),
 array([   0,    1,    2, ..., 5497, 5498, 5499]))
```
and therefore [our automl search code which uses `iloc`](https://github.com/alteryx/evalml/blob/main/evalml/automl/automl_search.py#L638) to make the dataframes for each split will work correctly.

For the TrainingValidationSplit, using `np.arange(X.shape[0])` as input to `train_test_split` as you've done here produces the same behavior: the result is the row number, and completely ignores the custom index, which is appropriate for `iloc`.

Great fix!